ui/src/sqlDialog.tsx

```diff
@@ -0,0 +1,88 @@
+import DownloadIcon from "@mui/icons-material/Download";
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import { EntityInstancesApiContext } from "apiContext";
+import React, { useContext, useEffect, useState } from "react";
+import { Dataset } from "./dataset";
+
+type SqlDialogProps = {
+  dataset: Dataset;
+};
+
+export default function SqlDialog(props: SqlDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [sql, setSql] = React.useState<string | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const api = useContext(EntityInstancesApiContext);
+
+  useEffect(() => {
+    if (open) {
+      const params = props.dataset.generateQueryParameters();
+      if (!!params) {
+        api
+          .generateDatasetSqlQuery({
+            entityName: props.dataset.entityName,
+            underlayName: props.dataset.underlayName,
+            generateDatasetSqlQueryRequest: {
+              entityDataset: params,
+            },
+          })
+          .then(
+            (res) => {
+              if (!!res?.query) {
+                setSql(res.query);
+                setError(null);
+              } else {
+                setError(new Error("Service returned an empty query"));
+              }
+            },
+            (error) => {
+              setError(error);
+            }
+          );
+      } else {
+        setError(new Error("No criteria have been selected"));
+      }
+    }
+  }, [api, props.dataset, open]);
+
+  return (
+    <>
+      <IconButton
```

Comment:
I'm more accustomed to a dialog component just being a dialog (not including a button to open/close it). This would allow the dialog to be opened from different contexts, e.g. from a menu but also from a button.

Reply:
It's a little weird because of how state works and it's not trivial to access to other components. What do you think about this update?